contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -1052,13 +1052,15 @@ contract VaultHub is PausableUntilWithRoles {
 
         uint256 unsettledLidoFees = _unsettledLidoFeesValue(_record);
         if (unsettledLidoFees > 0) {
-            uint256 _vaultBalance = _availableBalance(_vault);
-            if (_vaultBalance < unsettledLidoFees && _forceFullFeesSettlement) {
+            uint256 balance = Math256.min(_availableBalance(_vault), _totalValue(_record));
+            if (balance < unsettledLidoFees && _forceFullFeesSettlement) {
                 revert NoUnsettledLidoFeesShouldBeLeft(_vault, unsettledLidoFees);
             }
 
-            uint256 withdrawable = Math256.min(unsettledLidoFees, _vaultBalance);
-            _settleLidoFees(_vault, _record, _connection, withdrawable);
+            uint256 withdrawable = Math256.min(balance, unsettledLidoFees);
```

Comment:
looks a bit confusing from the first glance, prob would be good to branch on the `_forceFullFeesSettlement` at the first place 🙏